nixos/modules/services/networking/doh-proxy-rust.nix

```diff
@@ -0,0 +1,69 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.doh-proxy-rust;
+
+in {
+
+  options.services.doh-proxy-rust = {
+
+    enable = mkEnableOption "doh-proxy-rust";
+
+    flags = mkOption {
+      type = types.listOf types.str;
+      default = [];
+      example = literalExample [ "--server-address=9.9.9.9:53" ];
+      description = ''
+        A list of command-line flags to pass to doh-proxy.
+      '';
+    };
+
+  };
+
+  config = mkIf cfg.enable {
+
+    users.users.doh-proxy-rust = {
```

Comment:
Maybe you just want to use `DynamicUser` instead.

Reply:
This doesn't work together with `confinement`, but per the other comment, I've disabled `confinement` for now.

Reply:
Alright.